### PR TITLE
Demarche select validateur

### PR DESCRIPTION
**Pull request by XavierJp**
A merger avec modération

### CODE REVIEW

**Review by rdubigny — CHANGES_REQUESTED** [review]

File: src/components/ConfirmationModal.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import AriaModal from '@justfixnyc/react-aria-modal';
+
+const ConfirmationModal = ({
```
[the remark this answers is not in the document]
Comment (XavierJp) [reply src/components/ConfirmationModal.js:4]: Great but be careful as they are quite different (in term of CTA colour and behaviour)

File: src/components/form-sections/DemarcheSection/DemarcheSectionReadOnly.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import { findModifiedFields } from '../../../lib';
+
+export const DemarcheSectionReadOnly = ({ demarches }) => {
+  const { enrollment } = useContext(FormContext);
+  const { demarche: selectedDemarcheId } = enrollment;
+  const {
+    user: { roles },
+  } = useContext(UserContext);
+
+  const [modifiedFields, setModifiedFields] = useState(false);
```
[the remark this answers is not in the document]
Comment (XavierJp) [reply src/components/form-sections/DemarcheSection/DemarcheSectionReadOnly.js:17]: haha I definitely agree with you !

File: src/components/form-sections/DemarcheSection/DemarcheSectionSelect.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+export const DemarcheSectionSelect = ({ demarches }) => {
+  const { onChange, enrollment } = useContext(FormContext);
+  const { demarche: selectedDemarcheId } = enrollment;
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [confirmNewDemarcheId, setConfirmNewDemarcheId] = useState(false);
+
+  useEffect(() => {
+    // this is not great but state update goes like this
```
[the remark this answers is not in the document]
Comment (XavierJp) [reply src/components/form-sections/DemarcheSection/DemarcheSectionSelect.js:19]: My bad

File: src/components/form-sections/DemarcheSection/DemarcheSectionSelect.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+          >
+            {Object.keys(demarches).map(demarcheId => (
+              <option key={demarcheId} value={demarcheId}>
+                {get(demarches, demarcheId, {}).label}
+              </option>
+            ))}
+          </select>
+        </div>
+      </ScrollablePanel>
+      <DemarcheSectionSelectNotification
```
[the remark this answers is not in the document]
Comment (XavierJp) [reply src/components/form-sections/DemarcheSection/DemarcheSectionSelect.js:104]: oh oh tricky one :)

File: src/components/form-sections/DemarcheSection/DemarcheSectionSelectNotification.js [hunk 5]
```diff
@@ -7,54 +7,61 @@ const DemarcheSectionNotification = ({
...
-            votre cas.
-          </div>
-        )}
-      </div>
-    )}
-  </>
-);
+}) => {
+  const relevantDemarcheId =
+    Object.keys(demarches).indexOf(selectedDemarcheId) > -1 &&
```
Comment (rdubigny) [inline src/components/form-sections/DemarcheSection/DemarcheSectionSelectNotification.js:12]: This is quite known syntax for js dev but I find it hackish, what about lodash `has`:

`has(demarches, selectedDemarcheId)`

**Review by rdubigny — CHANGES_REQUESTED** [review]

File: src/components/ConfirmationModal.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rdubigny) [reply src/components/ConfirmationModal.js:4]: Duly noted. I'll keep this conversation un-resolved to remind me to do it.

**Pull request merged**